core/src/main/java/tech/pegasys/eth2signer/core/Eth2Signer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.eth2signer.core;
+
+import tech.pegasys.eth2signer.core.http.LogErrorHandler;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.handler.ResponseContentTypeHandler;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Eth2Signer implements Runnable {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String TEXT = HttpHeaderValues.TEXT_PLAIN.toString() + "; charset=utf-8";
+  private final Config config;
+
+  public Eth2Signer(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run() {
+    final HttpServerOptions serverOptions =
+        new HttpServerOptions()
+            .setPort(config.getHttpListenPort())
+            .setHost(config.getHttpListenHost())
+            .setReuseAddress(true)
+            .setReusePort(true);
+
+    final Vertx vertx = Vertx.vertx();
+    try {
+      final Router router = Router.router(vertx);
```

Comment:
This function is getting a bit big you split into a few functions perhaps

Reply:
done